src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -53,6 +56,268 @@ pub enum LinearActuatorMode {
+                    / Self::DISPLACEMENT_CU_IN,
+            )
+        } else {
+            AngularVelocity::default()
+        };
+
+        self.update_power_consumed(actuator_flow, actuator_pressure);
+        self.speed.update(context.delta(), new_speed);
+    }
+
+    fn update_power_consumed(&mut self, actuator_flow: VolumeRate, actuator_pressure: Pressure) {
+        // Horsepower = Pressure (PSIG) × Flow (GPM)/ 1714
+        let current_fluid_power = Power::new::<horsepower>(
+            actuator_pressure.get::<psi>() * actuator_flow.get::<gallon_per_minute>() / 1714.,
+        );
+
+        self.consumed_power = if self.is_active() {
+            (current_fluid_power * Self::HYD_TO_ELEC_POWER_EFFICIENCY)
+                .max(Power::new::<watt>(Self::MIN_STATIC_POWER_CONSUMPTION_WATT))
+        } else {
+            Power::default()
+        };
+    }
+
+    fn is_active(&self) -> bool {
+        self.should_activate_electrical_mode && self.is_powered
+    }
+
+    fn max_available_pressure(&self, accumulator_pressure: Pressure) -> Pressure {
+        if self.is_active() && accumulator_pressure.get::<psi>() > 100. {
+            Pressure::new::<psi>(Self::NOMINAL_MAX_PRESSURE_PSI)
+        } else {
+            Pressure::default()
+        }
+    }
+}
+impl SimulationElement for VariableSpeedPump {
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_powered = buses.is_powered(self.powered_by);
+    }
+
+    fn consume_power<T: ConsumePower>(&mut self, _: &UpdateContext, consumption: &mut T) {
+        consumption.consume_from_bus(self.powered_by, self.consumed_power);
+    }
+}
+
+#[derive(PartialEq, Copy, Clone)]
+struct LowPressureAccumulator {
+    pressure: LowPassFilter<Pressure>,
+    total_volume_to_actuator: Volume,
+}
+impl LowPressureAccumulator {
+    const MEAN_ACCUMULATOR_PRESSURE_PSI: f64 = 800.;
+    const STDEV_ACCUMULATOR_PRESSURE_PSI: f64 = 200.;
+
+    const MAX_ACCUMULATOR_PRESSURE_PSI: f64 = 1300.;
+
+    const PRESSURE_TIME_CONSTANT: Duration = Duration::from_millis(1000);
+
+    const REFILL_FLOW_GALLON_PER_S: f64 = 0.05;
+
+    fn new() -> Self {
+        let init_pressure_psi = random_from_normal_distribution(
+            Self::MEAN_ACCUMULATOR_PRESSURE_PSI,
+            Self::STDEV_ACCUMULATOR_PRESSURE_PSI,
+        );
+
+        Self {
+            pressure: LowPassFilter::<Pressure>::new_with_init_value(
+                Self::PRESSURE_TIME_CONSTANT,
+                Pressure::new::<psi>(init_pressure_psi),
+            ),
+            total_volume_to_actuator: Volume::default(),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        input_pressure: Pressure,
+        controller: &impl ElectroHydrostaticPowered,
+    ) {
+        let mut new_pressure =
+            if input_pressure > self.pressure.output() && controller.should_open_refill_valve() {
+                Pressure::new::<psi>(1800.)
```

Comment:
What does the `1800` mean here? This should probably be extracted into a constant.

Reply:
I don't know :D